spec/controllers/concerns/new_device_concern_spec.rb

```diff
@@ -5,20 +5,34 @@
     Class.new do
       include NewDeviceConcern
 
-      attr_reader :current_user, :user_session, :cookies
+      attr_reader :current_user, :user_session, :cookies, :authenticated
 
-      def initialize(current_user:, user_session:, cookies:)
+      def initialize(current_user:, user_session:, cookies:, authenticated:)
         @current_user = current_user
         @user_session = user_session
         @cookies = cookies
+        @authenticated = authenticated
+      end
+
+      def warden
+        warden = Class.new do
+          attr_accessor :authenticated
+
+          def authenticated?(_scope)
+            @authenticated
+          end
+        end.new
+        warden.authenticated = authenticated
```

Comment:
could use a `Data` class so that we get a constructor and a read attribute for "free"?
```suggestion
        warden = Data.define(:authenticated) do
          def authenticated?(_scope)
            authenticated
          end
        end.new(authenticated:)
```

Reply:
Ah, interesting! I'd looked into whether forms of `Struct` could be used for this, but hadn't considered `Data`.

Reply:
This is no longer necessary after f37b254, which returns to an implementation similar to the original without messing with Warden `authenticated?`.